### PR TITLE
PopupMenu options fix

### LINKED ISSUE #1535: Crash calling Dismiss() on PopupMenu if rootAnimator is null
## Overview
Calling Dismiss on PopupMenu with no Animator
## Expected Behavior
No crash
## Actual Behavior
Crash
## Steps to reproduce
Call Dismiss on PopupMenu with no Animator
The problem is inside Dismiss():

            if (rootAnimator.isInitialized)
            {
                rootAnimator.SetTrigger("Dehydrate");
            }
            else
            {
                gameObject.SetActive(false);
            }

## Unity Editor Version
2017.2.1.f1
## Mixed Reality Toolkit Release Version
Master, Dev, Stabilization

### PR DESCRIPTION
**Pull request by keveleigh**
Overview
---
The PopupMenu was consuming `OnInputDown` and `OnInputUp` events, while it seemed like it should consume (and other similar components were using) `OnInputClicked`. I reworded the example scene to be clearer, and also decreased the number of places states were being handled. The PopupMenu routes its "close" input handling through a button, but was also receiving its own input. Now, everything goes through the button.

Changes
---
- Fixes: #2040, fixes #1535

### CODE REVIEW

File: Assets/HoloToolkit-Examples/Input/Scripts/PopupMenu.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 namespace HoloToolkit.Unity.InputModule.Tests
 {
-    public class PopupMenu : MonoBehaviour, IInputHandler
+    public class PopupMenu : MonoBehaviour
```
Comment (StephenHodgson) [inline Assets/HoloToolkit-Examples/Input/Scripts/PopupMenu.cs]: Why are we removing the input handler?

Comment (keveleigh) [reply Assets/HoloToolkit-Examples/Input/Scripts/PopupMenu.cs]: The pop-up menu was only using the input handler to direct modal input in. It additionally had a handler for when the close button is pressed (which essentially did the same thing). Instead, I'm reusing the same event, which means that only the button needs to handle the input events.

**Comment by ivan2007** [comment]
Hi @keveleigh, I made some test with InputManagerTest scene:
1. When you open the modal window you are still able to move the movable objects (I don't know if this is outside the scope of PopupMenu).
2. The button actions of Shrink and Grow are set with reset.
3. If you place a CheckBoxSet inside the Popup window (IsModal is true) you can't change the state of the check boxes. The modal input stack of InputManager stop the event only if it is marked with used (see HandleEvent<T>).

**Comment by keveleigh** [comment]
Hey @ivan2007:

1. Yeah, at the moment it only consumes clicks. If we set it to be the `OverrideFocusedObject` instead of just modal, it should make it consume everything, which does appear to be the original intent.
2. I'm not sure what this is referring to.
3. The PopupMenu as provided in the Examples folder doesn't seem to be written in a way to really be used outside of the example. You might be able to rework it to allow input to get to the CheckBoxSet if that's what's focused. Sounds useful, and perhaps generalizing this component and graduating it to the main Toolkit UX components would be a good addition.

**Comment by ivan2007** [comment]
Hey @keveleigh,
2. I am referring to this:
buttonaction
perhaps it is already fixed.
3. Yes it will be very useful having a general component of this type. In any case PopupMenu is a good starting point.

**Pull request merged**